iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/provisioning/setup/ProvisioningCommon.java

```diff
@@ -415,54 +407,11 @@ public ProvisioningStatus registerDevice(ProvisioningDeviceClientTransportProtoc
-                    iothubsToFinishAt.contains(actualReprovisionedHub));
-    }
-
-    private void assertProvisionedDeviceWorks(String iothubUri, String deviceId) throws IOException, URISyntaxException
-    {
-        for (IotHubClientProtocol iotHubClientProtocol: IotHubClientProtocol.values())
-        {
-            if (iotHubClientProtocol == IotHubClientProtocol.MQTT_WS || iotHubClientProtocol == IotHubClientProtocol.AMQPS_WS)
-            {
-                // MQTT_WS/AMQP_WS does not support X509 because of a bug on service
-                continue;
-            }
-
-            DeviceClient deviceClient = new DeviceClient(iothubUri, deviceId, testInstance.securityProvider, iotHubClientProtocol);
-            deviceClient.close();
-        }
-    }
-
-    protected void assertProvisionedDeviceCapabilitiesAreExpected(DeviceCapabilities expectedDeviceCapabilities, String provisionedHubConnectionString) throws IOException, IotHubException, InterruptedException {
-        TwinClient twinClient = new TwinClient(provisionedHubConnectionString, TwinClientOptions.builder().httpReadTimeoutSeconds(HTTP_READ_TIMEOUT).build());
-
-        boolean deviceFoundInCorrectHub = false;
-        long startTime = System.currentTimeMillis();
-        Twin twin = new Twin();
-        while (!deviceFoundInCorrectHub)
-        {
-            twin = twinClient.get(testInstance.provisionedDeviceId);
-            deviceFoundInCorrectHub = twin.getCapabilities() != null;
-
-            Thread.sleep(3000);
-
-            if (System.currentTimeMillis() - startTime > QUERY_TIMEOUT_MILLISECONDS)
-            {
-                fail(CorrelationDetailsLoggingAssert.buildExceptionMessageDpsIndividualOrGroup("Timed out waiting for provisioned device " + testInstance.provisionedDeviceId + " to be found in expected hub", getHostName(provisioningServiceConnectionString), testInstance.groupId, testInstance.registrationId));
-            }
-        }
-
-        if (expectedDeviceCapabilities.isIotEdge())
-        {
-            assertTrue(CorrelationDetailsLoggingAssert.buildExceptionMessageDpsIndividualOrGroup("Provisioned device isn't edge device: " + testInstance.provisionedDeviceId, getHostName(provisioningServiceConnectionString), testInstance.groupId, testInstance.registrationId), twin.getCapabilities().isIotEdge());
-        }
-        else
-        {
-            assertTrue(CorrelationDetailsLoggingAssert.buildExceptionMessageDpsIndividualOrGroup("Provisioned device shouldn't be edge device " + testInstance.provisionedDeviceId, getHostName(provisioningServiceConnectionString), testInstance.groupId, testInstance.registrationId), twin.getCapabilities() == null || !twin.getCapabilities().isIotEdge());
-        }
+        DeviceClient deviceClient = new DeviceClient(iothubUri, deviceId, testInstance.securityProvider, IotHubClientProtocol.MQTT);
+        deviceClient.open(true);
```

Comment:
open is an I/o operation, right?

Reply:
Yes. I'm adding this operation because it was missing before. However, I'm limiting it to only one protocol (MQTT) to limit the number of device connections we make when verifying that the provisioning worked.